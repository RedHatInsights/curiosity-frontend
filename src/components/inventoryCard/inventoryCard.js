import React from 'react';
import PropTypes from 'prop-types';
import { TableVariant } from '@patternfly/react-table';
import { Bullseye, Card, CardActions, CardBody, CardFooter, CardHeader, CardHeaderMain } from '@patternfly/react-core';
import { TableToolbar } from '@redhat-cloud-services/frontend-components/TableToolbar';
import { useProductInventoryHostsConfig, useProductInventoryHostsQuery } from '../productView/productViewContext';
import { helpers } from '../../common';
import { connect, reduxSelectors } from '../../redux';
import Table from '../table/table';
import { Loader } from '../loader/loader';
import { MinHeight } from '../minHeight/minHeight';
import InventoryGuests from '../inventoryGuests/inventoryGuests';
import { inventoryCardHelpers } from './inventoryCardHelpers';
import Pagination from '../pagination/pagination';
import { ToolbarFieldDisplayName } from '../toolbar/toolbarFieldDisplayName';
import { paginationHelpers } from '../pagination/paginationHelpers';
import { RHSM_API_QUERY_SET_TYPES } from '../../services/rhsm/rhsmConstants';
import { translate } from '../i18n/i18n';
import { useGetInstancesInventory, useOnPageInstances, useOnColumnSortInstances } from './inventoryCardContext';

/**
 * Set up inventory cards. Expand filters with base settings.
 *
 * @param {object} props
 * @param {Node} props.cardActions
 * @param {boolean} props.isDisabled
 * @param {number} props.perPageDefault
 * @param {Function} props.t
 * @param {object} props.session
 * @param {Function} props.useGetInventory
 * @param {Function} props.useOnPage
 * @param {Function} props.useOnColumnSort
 * @param {Function} props.useProductInventoryConfig
 * @param {Function} props.useProductInventoryQuery
 * @fires onColumnSort
 * @fires onPage
 * @fires onUpdateInventoryData
 * @returns {Node}
 */
const InventoryCard = ({
  cardActions,
  isDisabled,
  perPageDefault,
  t,
  session,
  useGetInventory: useAliasGetInventory,
  useOnPage: useAliasOnPage,
  useOnColumnSort: useAliasOnColumnSort,
  useProductInventoryConfig: useAliasProductInventoryConfig,
  useProductInventoryQuery: useAliasProductInventoryQuery
}) => {
  const query = useAliasProductInventoryQuery();
  const onPage = useAliasOnPage();
  const onColumnSort = useAliasOnColumnSort();
  const { filters: filterInventoryData, settings } = useAliasProductInventoryConfig();
  const { error, fulfilled, pending, data = {} } = useAliasGetInventory({ isDisabled });
  const { data: listData = [], meta = {} } = data;

  if (isDisabled) {
    return (
      <Card className="curiosity-inventory-card__disabled">
        <CardBody>
          <Bullseye>{t('curiosity-inventory.tab', { context: 'disabled' })}</Bullseye>
        </CardBody>
      </Card>
    );
  }

  const itemCount = meta?.count;
  const updatedPerPage = query[RHSM_API_QUERY_SET_TYPES.LIMIT] || perPageDefault;
  const updatedOffset = query[RHSM_API_QUERY_SET_TYPES.OFFSET];
  const isLastPage = paginationHelpers.isLastPage(updatedOffset, updatedPerPage, itemCount);

  // Set an updated key to force refresh minHeight
  const minHeightContentRefreshKey =
    (fulfilled === true && itemCount < updatedPerPage && `bodyMinHeight-${updatedPerPage}-resize`) ||
    (fulfilled === true && isLastPage && `bodyMinHeight-${updatedPerPage}-resize`) ||
    (error === true && `bodyMinHeight-${updatedPerPage}-resize`) ||
    `bodyMinHeight-${updatedPerPage}`;

  /**
   * Render an inventory table.
   *
   * @returns {Node}
   */
  const renderTable = () => {
    let updatedColumnHeaders = [];

    const updatedRows = listData.map(({ ...cellData }) => {
      const { columnHeaders, cells } = inventoryCardHelpers.parseRowCellsListData({
        filters: inventoryCardHelpers.parseInventoryFilters({
          filters: filterInventoryData,
          onSort: onColumnSort,
          query
        }),
        cellData,
        session
      });

      updatedColumnHeaders = columnHeaders;
      const subscriptionManagerId = cellData?.subscriptionManagerId;
      const numberOfGuests = cellData?.numberOfGuests;
      let isSubTable;

      // Is there a subTable, callback, or attempt to determine, return boolean
      if (typeof settings?.hasSubTable === 'function') {
        isSubTable = settings.hasSubTable({ ...cellData }, { ...session });
      } else {
        isSubTable = numberOfGuests > 0 && subscriptionManagerId;
      }

      return {
        cells,
        expandedContent:
          (isSubTable && <InventoryGuests numberOfGuests={numberOfGuests} id={subscriptionManagerId} />) || undefined
      };
    });

    return (
      <Table
        borders
        variant={TableVariant.compact}
        className="curiosity-inventory-list"
        columnHeaders={updatedColumnHeaders}
        rows={updatedRows}
      />
    );
  };

  return (
    <Card className="curiosity-inventory-card">
      <MinHeight key="headerMinHeight" updateOnContent>
        <CardHeader className={(error && 'hidden') || ''} aria-hidden={error || false}>
          {cardActions}
          <CardActions className={(!itemCount && 'transparent') || ''} aria-hidden={!itemCount || false}>
            <Pagination
              isCompact
              isDisabled={pending || error}
              itemCount={itemCount}
              offset={updatedOffset}
              onPage={onPage}
              onPerPage={onPage}
              perPage={updatedPerPage}
            />
          </CardActions>
        </CardHeader>
      </MinHeight>
      <MinHeight key={minHeightContentRefreshKey} updateOnContent>
        <CardBody>
          <div className={(error && 'blur') || (pending && 'fadein') || ''}>
            {pending && (
              <Loader
                variant="table"
                tableProps={{
                  className: 'curiosity-inventory-list',
                  colCount: filterInventoryData?.length || (listData?.[0] && Object.keys(listData[0]).length) || 1,
                  colWidth:
                    (filterInventoryData?.length && filterInventoryData.map(({ cellWidth }) => cellWidth)) || [],
                  rowCount: listData?.length || updatedPerPage,
                  variant: TableVariant.compact
                }}
              />
            )}
            {!pending && renderTable()}
          </div>
        </CardBody>
      </MinHeight>
      <MinHeight key="footerMinHeight" updateOnContent>
        <CardFooter
          className={(error && 'hidden') || (!itemCount && 'transparent') || ''}
          aria-hidden={error || !itemCount || false}
        >
          <TableToolbar isFooter>
            <Pagination
              dropDirection="up"
              isDisabled={pending || error}
              itemCount={itemCount}
              offset={updatedOffset}
              onPage={onPage}
              onPerPage={onPage}
              perPage={updatedPerPage}
            />
          </TableToolbar>
        </CardFooter>
      </MinHeight>
    </Card>
  );
};

/**
 * Prop types.
 *
 * @type {{useOnPage: Function, t: Function, session: object, perPageDefault: number, isDisabled: boolean,
 *     useProductInventoryConfig: Function, useGetInventory: Function, useOnColumnSort: Function,
 *     useProductInventoryQuery: Function}}
 */
InventoryCard.propTypes = {
  cardActions: PropTypes.node,
  isDisabled: PropTypes.bool,
  perPageDefault: PropTypes.number,
  session: PropTypes.object,
  t: PropTypes.func,
  useGetInventory: PropTypes.func,
  useOnPage: PropTypes.func,
  useOnColumnSort: PropTypes.func,
  useProductInventoryConfig: PropTypes.func,
  useProductInventoryQuery: PropTypes.func
};

/**
 * Default props.
 *
 * @type {{useOnPage: Function, t: Function, session: object, perPageDefault: number, isDisabled: boolean,
 *     useProductInventoryConfig: Function, useGetInventory: Function, useOnColumnSort: Function,
 *     useProductInventoryQuery: Function}}
 */
InventoryCard.defaultProps = {
  cardActions: (
    <CardHeaderMain>
      <ToolbarFieldDisplayName />
    </CardHeaderMain>
  ),
  isDisabled: helpers.UI_DISABLED_TABLE_INSTANCES,
  perPageDefault: 10,
  session: {},
  t: translate,
  useGetInventory: useGetInstancesInventory,
  useOnPage: useOnPageInstances,
  useOnColumnSort: useOnColumnSortInstances,
  useProductInventoryConfig: useProductInventoryHostsConfig,
  useProductInventoryQuery: useProductInventoryHostsQuery
};

/**
 * Create a selector from applied state, props.
 *
 * @type {Function}
 */
const makeMapStateToProps = reduxSelectors.user.makeUserSession();

const ConnectedInventoryCard = connect(makeMapStateToProps)(InventoryCard);

export { ConnectedInventoryCard as default, ConnectedInventoryCard, InventoryCard };
