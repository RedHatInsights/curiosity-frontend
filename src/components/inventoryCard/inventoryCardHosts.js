import React from 'react';
import PropTypes from 'prop-types';
import { useGetHostsInventory, useOnColumnSortHosts } from './inventoryCardContext';
import InventoryCard from './inventoryCard';
import { helpers } from '../../common';

/**
 * @memberof InventoryCard
 * @module InventoryCardHosts
 */

/**
 * A hosts' system inventory component.
 *
 * @param {object} props
 * @param {boolean} props.isDisabled
 * @param {Function} props.useGetInventory
 * @param {Function} props.useOnColumnSort
 * @returns {React.ReactNode}
 */
const InventoryCardHosts = ({ ...props }) => <InventoryCard {...props} />;

/**
 * Prop types.
 *
 * @type {{isDisabled: boolean, useGetInventory: Function, useOnColumnSort: Function}}
 */
InventoryCardHosts.propTypes = {
  isDisabled: PropTypes.bool,
  useGetInventory: PropTypes.func,
  useOnColumnSort: PropTypes.func
};

/**
 * Default props.
 *
 * @type {{isDisabled: boolean, useGetInventory: Function, useOnColumnSort: Function}}
 */
InventoryCardHosts.defaultProps = {
  isDisabled: helpers.UI_DISABLED_TABLE_HOSTS,
  useGetInventory: useGetHostsInventory,
  useOnColumnSort: useOnColumnSortHosts
};

export { InventoryCardHosts as default, InventoryCardHosts };
