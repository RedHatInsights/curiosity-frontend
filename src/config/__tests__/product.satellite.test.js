import { config } from '../product.satellite';
import { parseRowCellsListData } from '../../components/inventoryCard/inventoryCardHelpers';
import {
  RHSM_API_QUERY_INVENTORY_SORT_DIRECTION_TYPES as SORT_DIRECTION_TYPES,
  RHSM_API_QUERY_SET_TYPES
} from '../../services/rhsm/rhsmConstants';

describe('Product Satellite config', () => {
  it('should apply hosts inventory configuration', () => {
    const { initialInventoryFilters: initialFilters, inventoryHostsQuery: inventoryQuery } = config;

    const inventoryData = {
      displayName: 'lorem',
      inventoryId: 'lorem inventory id',
      hardwareType: 'ipsum',
      measurementType: null,
      numberOfGuests: 3,
      sockets: 10,
      cores: 12,
      lastSeen: '2022-01-01T00:00:00.000Z',
      loremIpsum: 'hello world'
    };

    const filteredInventoryData = parseRowCellsListData({
      filters: initialFilters,
      cellData: inventoryData
    });

    expect(filteredInventoryData).toMatchSnapshot('filtered');

    const fallbackInventoryData = {
      ...inventoryData,
      inventoryId: null,
      lastSeen: null,
      cloudProvider: 'dolor sit'
    };

    const fallbackFilteredInventoryData = parseRowCellsListData({
      filters: initialFilters,
      cellData: fallbackInventoryData
    });

    expect(fallbackFilteredInventoryData).toMatchSnapshot('filtered, fallback display');

    expect(inventoryQuery[RHSM_API_QUERY_SET_TYPES.DIRECTION] === SORT_DIRECTION_TYPES.DESCENDING).toBe(true);
  });

  it('should apply guest inventory configuration', () => {
    const { initialGuestsFilters: initialFilters } = config;

    const guestsData = {
      displayName: 'lorem',
      inventoryId: 'lorem inventory id',
      subscriptionManagerId: 'lorem subscription id',
      lastSeen: '2022-01-01T00:00:00.000Z',
      loremIpsum: 'hello world'
    };

    const filteredGuestsData = parseRowCellsListData({
      filters: initialFilters,
      cellData: guestsData
    });

    expect(filteredGuestsData).toMatchSnapshot('filtered');

    const filteredGuestsDataAuthorized = parseRowCellsListData({
      filters: initialFilters,
      cellData: guestsData,
      session: { authorized: { inventory: true } }
    });

    expect(filteredGuestsDataAuthorized).toMatchSnapshot('filtered, authorized');
  });
});
