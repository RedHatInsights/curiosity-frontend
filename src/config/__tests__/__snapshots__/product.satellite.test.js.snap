// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Satellite config should apply an inventory configuration: filtered 1`] = `
{
  "cells": [
    {
      "title": "lorem ipsum",
    },
    {
      "title": "--",
    },
    {
      "title": <React.Fragment>
        t(curiosity-inventory.label, {"context":"category"})
         
        
      </React.Fragment>,
    },
    {
      "title": 200,
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header_number_of, {"context":"guests"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"category"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "Sockets": {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "value": 200,
    },
    "display_name": {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "value": "lorem ipsum",
    },
    "last_seen": {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
  },
}
`;

exports[`Product Satellite config should apply an inventory configuration: filtered, NOT authorized 1`] = `
{
  "cells": [
    {
      "title": "lorem ipsum",
    },
    {
      "title": "--",
    },
    {
      "title": <React.Fragment>
        t(curiosity-inventory.label, {"context":"category"})
         
        
      </React.Fragment>,
    },
    {
      "title": 200,
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header_number_of, {"context":"guests"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"category"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "Sockets": {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "value": 200,
    },
    "display_name": {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "value": "lorem ipsum",
    },
    "instance_id": {
      "title": "t(curiosity-inventory.header_instance, {"context":"id"})",
      "value": "XXXX-XXXX-XXXXX-XXXXX",
    },
    "last_seen": {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
  },
}
`;

exports[`Product Satellite config should apply an inventory configuration: filtered, authorized 1`] = `
{
  "cells": [
    {
      "title": <Button
        component="a"
        href="/insights/inventory/XXXX-XXXX-XXXXX-XXXXX/"
        isInline={true}
        variant="link"
      >
        lorem ipsum
      </Button>,
    },
    {
      "title": "--",
    },
    {
      "title": <React.Fragment>
        t(curiosity-inventory.label, {"context":"category"})
         
        
      </React.Fragment>,
    },
    {
      "title": 200,
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header_number_of, {"context":"guests"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"category"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "Sockets": {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "value": 200,
    },
    "display_name": {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "value": "lorem ipsum",
    },
    "instance_id": {
      "title": "t(curiosity-inventory.header_instance, {"context":"id"})",
      "value": "XXXX-XXXX-XXXXX-XXXXX",
    },
    "last_seen": {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
  },
}
`;

exports[`Product Satellite config should apply an inventory configuration: filtered, fallback display 1`] = `
{
  "cells": [
    {
      "title": "lorem ipsum",
    },
    {
      "title": "--",
    },
    {
      "title": <React.Fragment>
        t(curiosity-inventory.label, {"context":"category"})
         
        <Label
          color="purple"
        >
          t(curiosity-inventory.label_cloud_provider, {"context":"dolor sit"})
        </Label>
      </React.Fragment>,
    },
    {
      "title": 200,
    },
    {
      "title": "",
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header_number_of, {"context":"guests"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"category"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "Sockets": {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "value": 200,
    },
    "cloud_provider": {
      "title": "t(curiosity-inventory.header_cloud, {"context":"provider"})",
      "value": "dolor sit",
    },
    "display_name": {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "value": "lorem ipsum",
    },
    "instance_id": {
      "title": "t(curiosity-inventory.header_instance, {"context":"id"})",
      "value": null,
    },
    "last_seen": {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "value": null,
    },
  },
}
`;

exports[`Product Satellite config should apply an inventory configuration: filtered, infinite 1`] = `
{
  "cells": [
    {
      "title": "lorem ipsum",
    },
    {
      "title": "--",
    },
    {
      "title": <React.Fragment>
        t(curiosity-inventory.label, {"context":"category"})
         
        
      </React.Fragment>,
    },
    {
      "title": 200,
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header_number_of, {"context":"guests"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"category"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "Sockets": {
      "title": "t(curiosity-inventory.header, {"context":"Sockets"})",
      "value": 200,
    },
    "display_name": {
      "title": "t(curiosity-inventory.header_display, {"context":"name"})",
      "value": "lorem ipsum",
    },
    "has_infinite_quantity": {
      "title": "t(curiosity-inventory.header_has_infinite, {"context":"quantity"})",
      "value": false,
    },
    "last_seen": {
      "title": "t(curiosity-inventory.header_last, {"context":"seen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
  },
}
`;

exports[`Product Satellite config should apply graph configuration: filters 1`] = `
{
  "filtersSettings": [
    {
      "settings": {
        "actions": [
          {
            "id": "granularity",
            "position": "right",
          },
        ],
        "color": "#06c",
        "fill": "#8bc1f7",
        "groupMetric": [
          "Sockets",
        ],
        "isDisabledLegendClick": true,
        "isMetricDisplay": false,
        "isMultiMetric": true,
        "isStandalone": undefined,
        "metric": undefined,
        "metrics": [
          {
            "chartType": "area",
            "color": "#06c",
            "fill": "#8bc1f7",
            "id": "Sockets_physical",
            "isCapacity": false,
            "isStacked": true,
            "isThreshold": false,
            "isToolbarFilter": true,
            "metric": "Sockets",
            "query": {
              "category": "physical",
            },
            "stroke": "#06c",
            "strokeWidth": 2,
          },
          {
            "chartType": "area",
            "color": "#009596",
            "fill": "#a2d9d9",
            "id": "Sockets_virtual",
            "isCapacity": false,
            "isStacked": true,
            "isThreshold": false,
            "isToolbarFilter": true,
            "metric": "Sockets",
            "query": {
              "category": "virtual",
            },
            "stroke": "#009596",
            "strokeWidth": 2,
          },
          {
            "chartType": "area",
            "color": "#5752d1",
            "fill": "#b2b0ea",
            "id": "Sockets_cloud",
            "isCapacity": false,
            "isStacked": true,
            "isThreshold": false,
            "isToolbarFilter": true,
            "metric": "Sockets",
            "query": {
              "category": "cloud",
            },
            "stroke": "#5752d1",
            "strokeWidth": 2,
          },
        ],
        "padding": {
          "bottom": 75,
          "left": 75,
          "right": 45,
          "top": 45,
        },
        "query": {
          "category": "physical",
        },
        "stringId": "Sockets_physical",
        "stroke": "#06c",
      },
    },
  ],
}
`;

exports[`Product Satellite config should apply graph configuration: settings 1`] = `
{
  "actions": [
    {
      "id": "granularity",
      "position": "right",
    },
  ],
  "isDisabledLegendClick": true,
}
`;

exports[`Product Satellite config should apply guest inventory configuration: filtered 1`] = `
{
  "cells": [
    {
      "title": "lorem",
    },
    {
      "title": "lorem inventory id",
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header, {"context":"guestsDisplayName"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "displayName": {
      "title": "t(curiosity-inventory.header, {"context":"displayName"})",
      "value": "lorem",
    },
    "inventoryId": {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "value": "lorem inventory id",
    },
    "lastSeen": {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
    "loremIpsum": {
      "title": "t(curiosity-inventory.header, {"context":"loremIpsum"})",
      "value": "hello world",
    },
    "subscriptionManagerId": {
      "title": "t(curiosity-inventory.header, {"context":"subscriptionManagerId"})",
      "value": "lorem subscription id",
    },
  },
}
`;

exports[`Product Satellite config should apply guest inventory configuration: filtered, NOT authorized 1`] = `
{
  "cells": [
    {
      "title": "lorem",
    },
    {
      "title": "lorem inventory id",
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header, {"context":"guestsDisplayName"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "displayName": {
      "title": "t(curiosity-inventory.header, {"context":"displayName"})",
      "value": "lorem",
    },
    "inventoryId": {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "value": "lorem inventory id",
    },
    "lastSeen": {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
    "loremIpsum": {
      "title": "t(curiosity-inventory.header, {"context":"loremIpsum"})",
      "value": "hello world",
    },
    "subscriptionManagerId": {
      "title": "t(curiosity-inventory.header, {"context":"subscriptionManagerId"})",
      "value": "lorem subscription id",
    },
  },
}
`;

exports[`Product Satellite config should apply guest inventory configuration: filtered, authorized 1`] = `
{
  "cells": [
    {
      "title": <Button
        component="a"
        href="/insights/inventory/lorem inventory id/"
        isInline={true}
        variant="link"
      >
        lorem
      </Button>,
    },
    {
      "title": "lorem inventory id",
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header, {"context":"guestsDisplayName"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "displayName": {
      "title": "t(curiosity-inventory.header, {"context":"displayName"})",
      "value": "lorem",
    },
    "inventoryId": {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "value": "lorem inventory id",
    },
    "lastSeen": {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
    "loremIpsum": {
      "title": "t(curiosity-inventory.header, {"context":"loremIpsum"})",
      "value": "hello world",
    },
    "subscriptionManagerId": {
      "title": "t(curiosity-inventory.header, {"context":"subscriptionManagerId"})",
      "value": "lorem subscription id",
    },
  },
}
`;

exports[`Product Satellite config should apply guest inventory configuration: filtered, missing inventory id 1`] = `
{
  "cells": [
    {
      "title": "lorem",
    },
    {
      "title": "",
    },
    {
      "title": <DateFormat
        date="2022-01-01T00:00:00.000Z"
      />,
    },
  ],
  "columnHeaders": [
    {
      "title": "t(curiosity-inventory.header, {"context":"guestsDisplayName"})",
      "transforms": [],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "transforms": [
        [Function],
      ],
    },
    {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "transforms": [
        [Function],
      ],
    },
  ],
  "data": {
    "displayName": {
      "title": "t(curiosity-inventory.header, {"context":"displayName"})",
      "value": "lorem",
    },
    "inventoryId": {
      "title": "t(curiosity-inventory.header, {"context":"inventoryId"})",
      "value": "",
    },
    "lastSeen": {
      "title": "t(curiosity-inventory.header, {"context":"lastSeen"})",
      "value": "2022-01-01T00:00:00.000Z",
    },
    "loremIpsum": {
      "title": "t(curiosity-inventory.header, {"context":"loremIpsum"})",
      "value": "hello world",
    },
    "subscriptionManagerId": {
      "title": "t(curiosity-inventory.header, {"context":"subscriptionManagerId"})",
      "value": "lorem subscription id",
    },
  },
}
`;

exports[`Product Satellite config should apply toolbar configuration: filters 1`] = `
[
  {
    "id": "sla",
  },
  {
    "id": "usage",
    "selected": true,
  },
  {
    "id": "category",
  },
]
`;
